src/main.cpp

```diff
@@ -526,12 +548,21 @@ int main(int argc, char** argv)
     sigaction(SIGINT, &act, nullptr);
     sigaction(SIGALRM, &act, nullptr);
 
+    vmi_pid_t injected_pid = 0;
     if (injection_pid > 0 && inject_file)
     {
         PRINT_DEBUG("Starting injection with PID %i(%i) for %s\n", injection_pid, injection_thread, inject_file);
-        int ret = drakvuf->inject_cmd(injection_pid, injection_thread, inject_file, inject_cwd, injection_method, output, binary_path, target_process, injection_timeout, injection_global_search, args_count, args);
-        if (!ret)
+        injector_status_t ret = drakvuf->inject_cmd(injection_pid, injection_thread, inject_file, inject_cwd, injection_method, output, binary_path, target_process, injection_timeout, injection_global_search, args_count, args, &injected_pid);
+        switch (ret)
+        {
+        case INJECTOR_FAILED_WITH_ERROR_CODE:
+            rc = 0;
```

Comment:
For a fall-through you should add an explicit comment saying `/* fall-through */`